src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java

```diff
@@ -371,5 +372,56 @@ public CloudSdk build() {
           startListeners, runDevAppServerWaitSeconds);
     }
 
+    /**
+     * Attempt to find the Google Cloud SDK in various places.
+     * 
+     * @return the path to the root of the Google Cloud SDK
+     * @throws AppEngineException if not found
+     */
+    @Nonnull
+    private Path discoverSdkPath() {
+      for (CloudSdkResolver resolver : getResolvers()) {
```

Comment:
I would like to see some kind of order here so that the ordered queried resolvers are, for example, in Eclipse's case:
1) Context resolver,
2) Preference store resolver,
3) PathResolver (we should rename this to AutoDiscoveryResolver or something else),
4) CloudSdkPrompter,
etc.